### PR TITLE
Split `handler.py` file

### LINKED ISSUE #320: Trainer: Split handler and logic into different file
`handler.py` is too long and needs refactoring.

### PR DESCRIPTION
**Pull request by emcastillo**
Closes #320

### CODE REVIEW

**Comment by emcastillo** [comment]
PTAL!

**Review by kmaehashi — CHANGES_REQUESTED** [review]

File: pytorch_pfn_extras/handler/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+from pytorch_pfn_extras.handler._handler import BaseHandler, Handler  # NOQA
+from pytorch_pfn_extras.handler._logic import BaseLogic, Logic  # NOQA
```
[the remark this answers is not in the document]
Comment (emcastillo) [reply pytorch_pfn_extras/handler/__init__.py:2]: Good catch, thanks!!!

File: pytorch_pfn_extras/handler/__init__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+from pytorch_pfn_extras.handler._handler import BaseHandler, Handler  # NOQA
```
Comment (kmaehashi) [inline pytorch_pfn_extras/handler/__init__.py:1]: Sorry I mistakenly removed the `BaseLogic, Logic` import in my previous suggestion 😅 

```suggestion
from pytorch_pfn_extras.handler._handler import BaseHandler, Handler  # NOQA
from pytorch_pfn_extras.handler._logic import BaseLogic, Logic  # NOQA
```

**Review by kmaehashi — CHANGES_REQUESTED** [review]

File: pytorch_pfn_extras/handler/__init__.py (hunk 2 above)
Comment (emcastillo) [reply pytorch_pfn_extras/handler/__init__.py:1]: sorry, it was my fault for not double-checking 😥

File: pytorch_pfn_extras/handler/__init__.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+from pytorch_pfn_extras.handler._logic import BaseLogic, Logic  # NOQA
+
+# Deprecated, only imported for backward compatibility
+from pytorch_pfn_extras.handler._logic import torch_autocast
```
Comment (kmaehashi) [inline pytorch_pfn_extras/handler/__init__.py:5]:
```suggestion
from pytorch_pfn_extras.handler._logic import torch_autocast  # NOQA
```

**Pull request merged**